validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java

```diff
@@ -111,6 +119,27 @@ public Optional<Eth1Address> getFeeRecipient(final BLSPublicKey publicKey) {
         .orElse(defaultFeeRecipient);
   }
 
+  // Cannot set a fee recipient if the key is specified in the configuration file
+  // Cannot set fee recipient to 0x00
+  public void setFeeRecipient(final BLSPublicKey publicKey, final Eth1Address eth1Address)
+      throws SetFeeRecipientException {
+    if (eth1Address.equals(Eth1Address.ZERO)) {
+      throw new IllegalArgumentException("Cannot set fee recipient to 0x00 address.");
+    }
+    if (validatorIndexProvider.isEmpty()
+        || !validatorIndexProvider.get().containsPublicKey(publicKey)) {
+      throw new IllegalArgumentException(
+          "Validator public key not found when attempting to set fee recipient.");
```

Comment:
should this just be SetFeeRecipientException also?

Reply:
Yeah I think that makes sense.